File: VehicleTracker/VehicleTracker/Ship Tracker/Annotation.m
//
//  Annotation.m
//  MCP
//
//  Created by Nascenia on 6/1/15.
//  Copyright (c) 2015 Nascenia. All rights reserved.
//


#import "Annotation.h"

@implementation Annotation

@synthesize coordinate=_coordinate;

- (id)initWithCoordinate:(CLLocationCoordinate2D)coordinate forTitle:(NSString *)title forSubTitle:(NSString *)territory {
    self = [super init];
    
    if (self != nil) {
        _coordinate = coordinate;
        _name = title;
        _territory = territory;
    }
    
    return self;
}

-(CLLocation *) getLocation {
    return [[CLLocation alloc] initWithLatitude: _coordinate.latitude longitude: _coordinate.longitude];
}


//Add a Callout
- (NSString*) title {
    return self.name;
}

//Add a Callout
- (NSString*) subtitle {
    return self.territory;
}

@end

File: MCP - Swift/MCP/Annotation.swift
//
//  Annotation.swift
//  MCP
//
//  Created by Nascenia on 5/26/15.
//  Copyright (c) 2015 Nascenia. All rights reserved.
//

import UIKit
import MapKit

var ARROW_ANNOTATION : NSString = "ARROW_ANNOTATION"
var PIN_ANNOTATION  : NSString = "PIN_ANNOTATION"

class Annotation: NSObject, MKAnnotation {
    var currentLocation: CLLocationCoordinate2D
    var _title : String
    var subTitle : String
    var direction : CLLocationDirection!
    var typeOfAnnotation : String!
    
    init(coordinate: CLLocationCoordinate2D, title : String, subTitle : String) {
        self.currentLocation = coordinate
        self._title = title
        self.subTitle = subTitle
    }
    
    func getLocation() -> CLLocation {
        return CLLocation(latitude: coordinate.latitude, longitude: coordinate.longitude)
    }
    
    var coordinate: CLLocationCoordinate2D {
        return self.currentLocation
    }
    
    var title : String {
        return self._title
    }
    
    var subtitle : String {
        return self.subTitle
    }
}
